src/main/java/nextstep/security/oauth2/OAuth2RedirectFilter.java

```diff
@@ -0,0 +1,54 @@
+package nextstep.security.oauth2;
+
+import jakarta.servlet.*;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import nextstep.security.access.AntRequestMatcher;
+import nextstep.security.access.RequestMatcher;
+import nextstep.app.OAuth2RegistrationRepository;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.filter.GenericFilterBean;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.io.IOException;
+
+public class OAuth2RedirectFilter extends GenericFilterBean {
+
+    private final RequestMatcher requestMatcher;
+    private final OAuth2RegistrationRepository oAuth2RegistrationRepository;
+
+
+    public OAuth2RedirectFilter(final OAuth2RegistrationRepository oAuth2RegistrationRepository) {
+        this.requestMatcher = new AntRequestMatcher(HttpMethod.GET, "/oauth2/authorization/**");
+        this.oAuth2RegistrationRepository = oAuth2RegistrationRepository;
+    }
+
+    @Override
+    public void doFilter(final ServletRequest servletRequest, final ServletResponse servletResponse, final FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest httpServletRequest = (HttpServletRequest) servletRequest;
+        if (!requestMatcher.matches(httpServletRequest)) {
+            filterChain.doFilter(servletRequest, servletResponse);
+            return;
+        }
+        StringBuffer url = httpServletRequest.getRequestURL();
```

Comment:
```suggestion
        String url = httpServletRequest.getRequestURI();
```

RequestURL로 가져올 경우 전체 URL을 반환하기 때문에 `https://example.com/oauth2/authorization/google?state=some`와 같은 쿼리 스트링이 있을 경우 올바르게 동작하지 않을 수 있겠네요.
getRequestURI로 변경해보면 어떨까요?